src/components/Facility/AssetCreate.tsx

```diff
@@ -182,180 +194,137 @@ const AssetCreate = (props: AssetProps) => {
+        last_service: asset.last_service,
+        serviced_on: asset.last_service?.serviced_on,
+        notes: asset.last_service?.note,
+      });
     },
   });
 
-  const validateForm = () => {
-    const errors = { ...initError };
-    let invalidForm = false;
-    Object.keys(state.errors).forEach((field) => {
-      switch (field) {
-        case "name":
-          if (!name) {
-            errors[field] = "Asset name can't be empty";
-            invalidForm = true;
-          }
-          return;
-        case "is_working":
-          if (is_working === undefined) {
-            errors[field] = t("field_required");
-            invalidForm = true;
-          }
-          return;
-        case "location":
-          if (!location || location === "0" || location === "") {
-            errors[field] = "Select a location";
-            invalidForm = true;
-          }
-          return;
-        case "support_phone": {
-          if (!support_phone) {
-            errors[field] = t("field_required");
-            invalidForm = true;
-          }
-          // eslint-disable-next-line no-case-declarations
-          const checkTollFree = support_phone.startsWith("1800");
-          const supportPhoneSimple = support_phone
-            .replace(/[^0-9]/g, "")
-            .slice(2);
-          if (supportPhoneSimple.length != 10 && !checkTollFree) {
-            errors[field] = "Please enter valid phone number";
-            invalidForm = true;
-          } else if (
-            (support_phone.length < 10 || support_phone.length > 11) &&
-            checkTollFree
-          ) {
-            errors[field] = "Please enter valid phone number";
-            invalidForm = true;
-          }
-          return;
-        }
-        case "support_email":
-          if (support_email && !validateEmailAddress(support_email)) {
-            errors[field] = "Please enter valid email id";
-            invalidForm = true;
-          }
-          return;
-        case "last_serviced_on":
-          if (notes && !last_serviced_on) {
-            errors[field] = "Last serviced on date is require with notes";
-            invalidForm = true;
-          }
-          return;
-        default:
-          return;
-      }
-    });
-    if (invalidForm) {
-      dispatch({ type: "set_error", errors });
-      const firstError = Object.keys(errors).find((key) => errors[key]);
-      if (firstError) {
-        fieldRef[firstError].current?.scrollIntoView({
-          behavior: "smooth",
-          block: "center",
-        });
+  const AssetFormValidator = (form: AssetT): FormErrors<AssetT> => {
+    const errors: Partial<Record<keyof AssetT, FieldError>> = {}; // Initialize error object
+
+    errors.name = RequiredFieldValidator()(form.name);
+
+    if (form.is_working === undefined) {
+      errors.is_working = t("field_required");
+    }
+
+    if (!form.location || form.location === "0" || form.location === "") {
+      errors.location = "Select a location";
```

Comment:
Switch right of these to using translations as well, like in the line right above.

Add key value pair to en.json file. Some might already be present, so be sure to check first.